diff --git a/dist/cjs/sdk.js b/dist/cjs/sdk.js
index e5b1006629b14e828a4c653d7e66ee65334b3e8a..0ba10db3c7f89fcd9bcf556702ca48096d802274 100644
--- a/dist/cjs/sdk.js
+++ b/dist/cjs/sdk.js
@@ -1083,6 +1083,10 @@ class Client {
                 window.console.warn('Appwrite is using localStorage for session management. Increase your security by adding a custom domain as your API endpoint.');
                 window.localStorage.setItem('cookieFallback', cookieFallback);
             }
+            const setCookie = response.headers.getSetCookie();
+            if (data && typeof data === 'object' && setCookie) {
+              data.setCookie = setCookie;
+            }
             return data;
         });
     }
@@ -1559,6 +1563,7 @@ class Account {
         const name = params.name;
         const scopes = params.scopes;
         const expire = params.expire;
+        const secret = params.secret;
         if (typeof name === 'undefined') {
             throw new AppwriteException('Missing required parameter: "name"');
         }
@@ -1576,6 +1581,9 @@ class Account {
         if (typeof expire !== 'undefined') {
             payload['expire'] = expire;
         }
+        if (typeof secret !== 'undefined') {
+            payload['secret'] = secret;
+        }
         const uri = new URL(this.client.config.endpoint + apiPath);
         const apiHeaders = {
             'content-type': 'application/json',
diff --git a/dist/esm/sdk.js b/dist/esm/sdk.js
index d3878b9e8545559c8e27d48de852cd8ff623980f..5df326f86926d6e7380b2321fb5e705675c82ad2 100644
--- a/dist/esm/sdk.js
+++ b/dist/esm/sdk.js
@@ -1076,6 +1076,10 @@ class Client {
                 window.console.warn('Appwrite is using localStorage for session management. Increase your security by adding a custom domain as your API endpoint.');
                 window.localStorage.setItem('cookieFallback', cookieFallback);
             }
+            const setCookie = response.headers.getSetCookie();
+            if (data && typeof data === 'object' && setCookie) {
+              data.setCookie = setCookie;
+            }
             return data;
         });
     }
@@ -1552,6 +1556,7 @@ class Account {
         const name = params.name;
         const scopes = params.scopes;
         const expire = params.expire;
+        const secret = params.secret;
         if (typeof name === 'undefined') {
             throw new AppwriteException('Missing required parameter: "name"');
         }
@@ -1569,6 +1574,9 @@ class Account {
         if (typeof expire !== 'undefined') {
             payload['expire'] = expire;
         }
+        if (typeof secret !== 'undefined') {
+            payload['secret'] = secret;
+        }
         const uri = new URL(this.client.config.endpoint + apiPath);
         const apiHeaders = {
             'content-type': 'application/json',
