diff --git a/dist/index.mjs b/dist/index.mjs
index c4c3182bf11a6ed6d19e18dba935258ba4cc7402..ca04586bd12cafe83cf096e01fac7769b24a41a6 100644
--- a/dist/index.mjs
+++ b/dist/index.mjs
@@ -41524,7 +41524,7 @@ function resolvePluginConfig(pluginConfig, userConfig, viteEnv) {
 		experimental: { headersAndRedirectsDevModeSupport: pluginConfig.experimental?.headersAndRedirectsDevModeSupport }
 	};
 	const root = userConfig.root ? nodePath.resolve(userConfig.root) : process.cwd();
-	const prefixedEnv = vite.loadEnv(viteEnv.mode, root, ["CLOUDFLARE_", "WRANGLER_HYPERDRIVE_LOCAL_CONNECTION_STRING_"]);
+	const prefixedEnv = vite.loadEnv(viteEnv.mode, userConfig.envDir ? nodePath.resolve(root, userConfig.envDir) : root, ["CLOUDFLARE_", "WRANGLER_HYPERDRIVE_LOCAL_CONNECTION_STRING_"]);
 	Object.assign(process.env, prefixedEnv);
 	if (viteEnv.isPreview) return {
 		...shared,
