diff --git a/dist/api/index.d.mts b/dist/api/index.d.mts
index 62916273df7a8740005afa16dd4ed167aef81d05..9556b80645fd98bebbc46a42a6e02c359851369e 100644
--- a/dist/api/index.d.mts
+++ b/dist/api/index.d.mts
@@ -1721,9 +1721,9 @@ declare function getEndpoints<Option extends BetterAuthOptions>(ctx: Awaitable<A
       id: string;
       createdAt: Date;
       updatedAt: Date;
-      userId: string;
       providerId: string;
       accountId: string;
+      userId: string;
     }[]>;
     readonly deleteUserCallback: better_call0.StrictEndpoint<"/delete-user/callback", {
       method: "GET";
@@ -3709,9 +3709,9 @@ declare const router: <Option extends BetterAuthOptions>(ctx: AuthContext, optio
       id: string;
       createdAt: Date;
       updatedAt: Date;
-      userId: string;
       providerId: string;
       accountId: string;
+      userId: string;
     }[]>;
     readonly deleteUserCallback: better_call0.StrictEndpoint<"/delete-user/callback", {
       method: "GET";
diff --git a/dist/api/routes/account.d.mts b/dist/api/routes/account.d.mts
index 9157b8d91c37fa8beb593c7b79a38df74f182c81..eb04fcacf87b66d469c9c5b956ea8551f1b54b96 100644
--- a/dist/api/routes/account.d.mts
+++ b/dist/api/routes/account.d.mts
@@ -83,9 +83,9 @@ declare const listUserAccounts: better_call0.StrictEndpoint<"/list-accounts", {
   id: string;
   createdAt: Date;
   updatedAt: Date;
-  userId: string;
   providerId: string;
   accountId: string;
+  userId: string;
 }[]>;
 declare const linkSocialAccount: better_call0.StrictEndpoint<"/link-social", {
   method: "POST";
diff --git a/dist/client/lynx/index.d.mts b/dist/client/lynx/index.d.mts
index 3adeb2f3c4754d895207de4abe607a5c1a0bae77..695f52cc3dff340bf5bb79ea3b6c94c37a3c9bb0 100644
--- a/dist/client/lynx/index.d.mts
+++ b/dist/client/lynx/index.d.mts
@@ -66,19 +66,19 @@ declare function createAuthClient<Option extends BetterAuthClientOptions>(option
         onResponse: ((context: _better_fetch_fetch0.ResponseContext) => Promise<Response | void | _better_fetch_fetch0.ResponseContext> | Response | _better_fetch_fetch0.ResponseContext | void) | undefined;
       };
     })[];
-    priority?: RequestPriority | undefined;
-    cache?: RequestCache | undefined;
-    credentials?: RequestCredentials;
+    method: string;
     headers?: (HeadersInit & (HeadersInit | {
       accept: "application/json" | "text/plain" | "application/octet-stream";
       "content-type": "application/json" | "text/plain" | "application/x-www-form-urlencoded" | "multipart/form-data" | "application/octet-stream";
       authorization: "Bearer" | "Basic";
     })) | undefined;
+    redirect?: RequestRedirect | undefined;
+    priority?: RequestPriority | undefined;
+    cache?: RequestCache | undefined;
+    credentials?: RequestCredentials;
     integrity?: string | undefined;
     keepalive?: boolean | undefined;
-    method: string;
     mode?: RequestMode | undefined;
-    redirect?: RequestRedirect | undefined;
     referrer?: string | undefined;
     referrerPolicy?: ReferrerPolicy | undefined;
     signal?: (AbortSignal | null) | undefined;
diff --git a/dist/client/react/index.d.mts b/dist/client/react/index.d.mts
index d9d1eef8051f431b3fc4ea59356354b23b0f689a..f583c5d6aa4f22315b0b2c3ced293c93a6532697 100644
--- a/dist/client/react/index.d.mts
+++ b/dist/client/react/index.d.mts
@@ -67,19 +67,19 @@ declare function createAuthClient<Option extends BetterAuthClientOptions>(option
         onResponse: ((context: _better_fetch_fetch0.ResponseContext) => Promise<Response | void | _better_fetch_fetch0.ResponseContext> | Response | _better_fetch_fetch0.ResponseContext | void) | undefined;
       };
     })[];
-    priority?: RequestPriority | undefined;
-    cache?: RequestCache | undefined;
-    credentials?: RequestCredentials;
+    method: string;
     headers?: (HeadersInit & (HeadersInit | {
       accept: "application/json" | "text/plain" | "application/octet-stream";
       "content-type": "application/json" | "text/plain" | "application/x-www-form-urlencoded" | "multipart/form-data" | "application/octet-stream";
       authorization: "Bearer" | "Basic";
     })) | undefined;
+    redirect?: RequestRedirect | undefined;
+    priority?: RequestPriority | undefined;
+    cache?: RequestCache | undefined;
+    credentials?: RequestCredentials;
     integrity?: string | undefined;
     keepalive?: boolean | undefined;
-    method: string;
     mode?: RequestMode | undefined;
-    redirect?: RequestRedirect | undefined;
     referrer?: string | undefined;
     referrerPolicy?: ReferrerPolicy | undefined;
     signal?: (AbortSignal | null) | undefined;
diff --git a/dist/client/solid/index.d.mts b/dist/client/solid/index.d.mts
index 934d2146772e61b2d8ba44d23bc43e9874b22ed3..365aa17fc3eb004d0deed69991c338bffd13c40a 100644
--- a/dist/client/solid/index.d.mts
+++ b/dist/client/solid/index.d.mts
@@ -66,19 +66,19 @@ declare function createAuthClient<Option extends BetterAuthClientOptions>(option
         onResponse: ((context: _better_fetch_fetch0.ResponseContext) => Promise<Response | void | _better_fetch_fetch0.ResponseContext> | Response | _better_fetch_fetch0.ResponseContext | void) | undefined;
       };
     })[];
-    priority?: RequestPriority | undefined;
-    cache?: RequestCache | undefined;
-    credentials?: RequestCredentials;
+    method: string;
     headers?: (HeadersInit & (HeadersInit | {
       accept: "application/json" | "text/plain" | "application/octet-stream";
       "content-type": "application/json" | "text/plain" | "application/x-www-form-urlencoded" | "multipart/form-data" | "application/octet-stream";
       authorization: "Bearer" | "Basic";
     })) | undefined;
+    redirect?: RequestRedirect | undefined;
+    priority?: RequestPriority | undefined;
+    cache?: RequestCache | undefined;
+    credentials?: RequestCredentials;
     integrity?: string | undefined;
     keepalive?: boolean | undefined;
-    method: string;
     mode?: RequestMode | undefined;
-    redirect?: RequestRedirect | undefined;
     referrer?: string | undefined;
     referrerPolicy?: ReferrerPolicy | undefined;
     signal?: (AbortSignal | null) | undefined;
diff --git a/dist/client/svelte/index.d.mts b/dist/client/svelte/index.d.mts
index 449d73c4b8d6e656dcd455be98a4589779b87a92..cbd4dc45a2a2486e4875163c0144a17a4257ce8a 100644
--- a/dist/client/svelte/index.d.mts
+++ b/dist/client/svelte/index.d.mts
@@ -52,19 +52,19 @@ declare function createAuthClient<Option extends BetterAuthClientOptions>(option
         onResponse: ((context: _better_fetch_fetch0.ResponseContext) => Promise<Response | void | _better_fetch_fetch0.ResponseContext> | Response | _better_fetch_fetch0.ResponseContext | void) | undefined;
       };
     })[];
-    priority?: RequestPriority | undefined;
-    cache?: RequestCache | undefined;
-    credentials?: RequestCredentials;
+    method: string;
     headers?: (HeadersInit & (HeadersInit | {
       accept: "application/json" | "text/plain" | "application/octet-stream";
       "content-type": "application/json" | "text/plain" | "application/x-www-form-urlencoded" | "multipart/form-data" | "application/octet-stream";
       authorization: "Bearer" | "Basic";
     })) | undefined;
+    redirect?: RequestRedirect | undefined;
+    priority?: RequestPriority | undefined;
+    cache?: RequestCache | undefined;
+    credentials?: RequestCredentials;
     integrity?: string | undefined;
     keepalive?: boolean | undefined;
-    method: string;
     mode?: RequestMode | undefined;
-    redirect?: RequestRedirect | undefined;
     referrer?: string | undefined;
     referrerPolicy?: ReferrerPolicy | undefined;
     signal?: (AbortSignal | null) | undefined;
diff --git a/dist/client/vanilla.d.mts b/dist/client/vanilla.d.mts
index be335c9193f6d616f498b9284b75738092ed8718..f1aa057ed625674fcd43302030773d39b981f71e 100644
--- a/dist/client/vanilla.d.mts
+++ b/dist/client/vanilla.d.mts
@@ -50,19 +50,19 @@ declare function createAuthClient<Option extends BetterAuthClientOptions>(option
         onResponse: ((context: _better_fetch_fetch0.ResponseContext) => Promise<Response | void | _better_fetch_fetch0.ResponseContext> | Response | _better_fetch_fetch0.ResponseContext | void) | undefined;
       };
     })[];
-    priority?: RequestPriority | undefined;
-    cache?: RequestCache | undefined;
-    credentials?: RequestCredentials;
+    method: string;
     headers?: (HeadersInit & (HeadersInit | {
       accept: "application/json" | "text/plain" | "application/octet-stream";
       "content-type": "application/json" | "text/plain" | "application/x-www-form-urlencoded" | "multipart/form-data" | "application/octet-stream";
       authorization: "Bearer" | "Basic";
     })) | undefined;
+    redirect?: RequestRedirect | undefined;
+    priority?: RequestPriority | undefined;
+    cache?: RequestCache | undefined;
+    credentials?: RequestCredentials;
     integrity?: string | undefined;
     keepalive?: boolean | undefined;
-    method: string;
     mode?: RequestMode | undefined;
-    redirect?: RequestRedirect | undefined;
     referrer?: string | undefined;
     referrerPolicy?: ReferrerPolicy | undefined;
     signal?: (AbortSignal | null) | undefined;
diff --git a/dist/client/vue/index.d.mts b/dist/client/vue/index.d.mts
index 378e431b51ed40157a515fe3e8bb4d76456e0fb8..2dee39975bacfbaaf76750a46698be414ccede75 100644
--- a/dist/client/vue/index.d.mts
+++ b/dist/client/vue/index.d.mts
@@ -90,19 +90,19 @@ declare function createAuthClient<Option extends BetterAuthClientOptions>(option
         onResponse: ((context: _better_fetch_fetch0.ResponseContext) => Promise<Response | void | _better_fetch_fetch0.ResponseContext> | Response | _better_fetch_fetch0.ResponseContext | void) | undefined;
       };
     })[];
-    priority?: RequestPriority | undefined;
-    cache?: RequestCache | undefined;
-    credentials?: RequestCredentials;
+    method: string;
     headers?: (HeadersInit & (HeadersInit | {
       accept: "application/json" | "text/plain" | "application/octet-stream";
       "content-type": "application/json" | "text/plain" | "application/x-www-form-urlencoded" | "multipart/form-data" | "application/octet-stream";
       authorization: "Bearer" | "Basic";
     })) | undefined;
+    redirect?: RequestRedirect | undefined;
+    priority?: RequestPriority | undefined;
+    cache?: RequestCache | undefined;
+    credentials?: RequestCredentials;
     integrity?: string | undefined;
     keepalive?: boolean | undefined;
-    method: string;
     mode?: RequestMode | undefined;
-    redirect?: RequestRedirect | undefined;
     referrer?: string | undefined;
     referrerPolicy?: ReferrerPolicy | undefined;
     signal?: (AbortSignal | null) | undefined;
diff --git a/dist/db/schema.d.mts b/dist/db/schema.d.mts
index f143c831713f58469341e39ab88e4d9029a9aa63..71c0f8be9fe372f353fbcdff0824810447a2b545 100644
--- a/dist/db/schema.d.mts
+++ b/dist/db/schema.d.mts
@@ -5,7 +5,7 @@ import { BetterAuthPluginDBSchema, DBFieldAttribute } from "@better-auth/core/db
 //#region src/db/schema.d.ts
 declare function parseUserOutput<T extends User$1>(options: BetterAuthOptions, user: T): T;
 declare function parseSessionOutput<T extends Session$1>(options: BetterAuthOptions, session: T): T;
-declare function parseAccountOutput<T extends Account>(options: BetterAuthOptions, account: T): Omit<T, "idToken" | "accessToken" | "refreshToken" | "accessTokenExpiresAt" | "refreshTokenExpiresAt" | "password">;
+declare function parseAccountOutput<T extends Account>(options: BetterAuthOptions, account: T): Omit<T, "accessToken" | "refreshToken" | "idToken" | "accessTokenExpiresAt" | "refreshTokenExpiresAt" | "password">;
 declare function parseInputData<T extends Record<string, any>>(data: T, schema: {
   fields: Record<string, DBFieldAttribute>;
   action?: ("create" | "update") | undefined;
diff --git a/dist/db/with-hooks.mjs b/dist/db/with-hooks.mjs
index fdea8a35f028f3d7751302b82f97b6a250e491fc..cfeb3242a28036661c41f2293bbac529f30ef6a7 100644
--- a/dist/db/with-hooks.mjs
+++ b/dist/db/with-hooks.mjs
@@ -28,6 +28,7 @@ function getWithHooks(adapter, ctx) {
 			forceAllowId: true
 		});
 		if (customCreateFn?.fn) created = await customCreateFn.fn(created ?? actualData);
+		const currentAdapter = await getCurrentAdapter(adapter);
 		for (const { source, hooks } of hooksEntries) {
 			const toRun = hooks[model]?.create?.after;
 			if (toRun) await queueAfterTransactionHook(async () => {
@@ -35,7 +36,7 @@ function getWithHooks(adapter, ctx) {
 					[ATTR_HOOK_TYPE]: "create.after",
 					[ATTR_DB_COLLECTION_NAME]: model,
 					[ATTR_CONTEXT]: source
-				}, () => toRun(created, context));
+				}, () => toRun(created, context, currentAdapter));
 			});
 		}
 		return created;
diff --git a/dist/plugins/username/index.d.mts b/dist/plugins/username/index.d.mts
index 148cc431cbfacca4052774345c2c4dd4bffd8586..6b60c9065e7fcd87a4f44f6b6e96186fe640ba9b 100644
--- a/dist/plugins/username/index.d.mts
+++ b/dist/plugins/username/index.d.mts
@@ -227,14 +227,14 @@ declare const username: (options?: UsernameOptions | undefined) => {
           unique: true;
           returned: true;
           transform: {
-            input(value: _better_auth_core_db0.DBPrimitive): string | number | boolean | Date | Record<string, unknown> | unknown[] | null | undefined;
+            input(value: _better_auth_core_db0.DBPrimitive): string | number | boolean | unknown[] | Record<string, unknown> | Date | null | undefined;
           };
         };
         displayUsername: {
           type: "string";
           required: false;
           transform: {
-            input(value: _better_auth_core_db0.DBPrimitive): string | number | boolean | Date | Record<string, unknown> | unknown[] | null | undefined;
+            input(value: _better_auth_core_db0.DBPrimitive): string | number | boolean | unknown[] | Record<string, unknown> | Date | null | undefined;
           };
         };
       };
diff --git a/dist/plugins/username/schema.d.mts b/dist/plugins/username/schema.d.mts
index 5bfdc8b354c4eaca325550ac5ffd248b4953e773..1f6131c2d66f550d6a8ac76243fd926d8a03a64f 100644
--- a/dist/plugins/username/schema.d.mts
+++ b/dist/plugins/username/schema.d.mts
@@ -14,14 +14,14 @@ declare const getSchema: (normalizer: {
         unique: true;
         returned: true;
         transform: {
-          input(value: _better_auth_core_db0.DBPrimitive): string | number | boolean | Date | Record<string, unknown> | unknown[] | null | undefined;
+          input(value: _better_auth_core_db0.DBPrimitive): string | number | boolean | unknown[] | Record<string, unknown> | Date | null | undefined;
         };
       };
       displayUsername: {
         type: "string";
         required: false;
         transform: {
-          input(value: _better_auth_core_db0.DBPrimitive): string | number | boolean | Date | Record<string, unknown> | unknown[] | null | undefined;
+          input(value: _better_auth_core_db0.DBPrimitive): string | number | boolean | unknown[] | Record<string, unknown> | Date | null | undefined;
         };
       };
     };
diff --git a/dist/test-utils/test-instance.d.mts b/dist/test-utils/test-instance.d.mts
index 881f2cb4dc8399deee94fa97f30716a034aca6a2..f977ee21f2ebb3a9f0bf128b87c49972d5fae38b 100644
--- a/dist/test-utils/test-instance.d.mts
+++ b/dist/test-utils/test-instance.d.mts
@@ -1733,9 +1733,9 @@ declare function getTestInstance<O extends Partial<BetterAuthOptions>, C extends
       id: string;
       createdAt: Date;
       updatedAt: Date;
-      userId: string;
       providerId: string;
       accountId: string;
+      userId: string;
     }[]>;
     readonly deleteUserCallback: better_call0.StrictEndpoint<"/delete-user/callback", {
       method: "GET";
@@ -3735,9 +3735,9 @@ declare function getTestInstance<O extends Partial<BetterAuthOptions>, C extends
       id: string;
       createdAt: Date;
       updatedAt: Date;
-      userId: string;
       providerId: string;
       accountId: string;
+      userId: string;
     }[]>;
     readonly deleteUserCallback: better_call0.StrictEndpoint<"/delete-user/callback", {
       method: "GET";
@@ -5740,9 +5740,9 @@ declare function getTestInstance<O extends Partial<BetterAuthOptions>, C extends
           id: string;
           createdAt: Date;
           updatedAt: Date;
-          userId: string;
           providerId: string;
           accountId: string;
+          userId: string;
         }[]>;
         readonly deleteUserCallback: better_call0.StrictEndpoint<"/delete-user/callback", {
           method: "GET";
@@ -7742,9 +7742,9 @@ declare function getTestInstance<O extends Partial<BetterAuthOptions>, C extends
           id: string;
           createdAt: Date;
           updatedAt: Date;
-          userId: string;
           providerId: string;
           accountId: string;
+          userId: string;
         }[]>;
         readonly deleteUserCallback: better_call0.StrictEndpoint<"/delete-user/callback", {
           method: "GET";
@@ -8068,19 +8068,19 @@ declare function getTestInstance<O extends Partial<BetterAuthOptions>, C extends
           onResponse: ((context: _better_fetch_fetch0.ResponseContext) => Promise<Response | void | _better_fetch_fetch0.ResponseContext> | Response | _better_fetch_fetch0.ResponseContext | void) | undefined;
         };
       })[];
-      priority?: RequestPriority | undefined;
-      cache?: RequestCache | undefined;
-      credentials?: RequestCredentials;
+      method: string;
       headers?: (HeadersInit & (HeadersInit | {
         accept: "application/json" | "text/plain" | "application/octet-stream";
         "content-type": "application/json" | "text/plain" | "application/x-www-form-urlencoded" | "multipart/form-data" | "application/octet-stream";
         authorization: "Bearer" | "Basic";
       })) | undefined;
+      redirect?: RequestRedirect | undefined;
+      priority?: RequestPriority | undefined;
+      cache?: RequestCache | undefined;
+      credentials?: RequestCredentials;
       integrity?: string | undefined;
       keepalive?: boolean | undefined;
-      method: string;
       mode?: RequestMode | undefined;
-      redirect?: RequestRedirect | undefined;
       referrer?: string | undefined;
       referrerPolicy?: ReferrerPolicy | undefined;
       signal?: (AbortSignal | null) | undefined;
@@ -9818,9 +9818,9 @@ declare function getTestInstance<O extends Partial<BetterAuthOptions>, C extends
           id: string;
           createdAt: Date;
           updatedAt: Date;
-          userId: string;
           providerId: string;
           accountId: string;
+          userId: string;
         }[]>;
         readonly deleteUserCallback: better_call0.StrictEndpoint<"/delete-user/callback", {
           method: "GET";
@@ -11820,9 +11820,9 @@ declare function getTestInstance<O extends Partial<BetterAuthOptions>, C extends
           id: string;
           createdAt: Date;
           updatedAt: Date;
-          userId: string;
           providerId: string;
           accountId: string;
+          userId: string;
         }[]>;
         readonly deleteUserCallback: better_call0.StrictEndpoint<"/delete-user/callback", {
           method: "GET";
